kge/model/kge_model.py

```diff
@@ -454,10 +454,11 @@ def prepare_job(self, job: "Job", **kwargs):
         self._entity_embedder.prepare_job(job, **kwargs)
         self._relation_embedder.prepare_job(job, **kwargs)
 
-        def append_num_parameter(job, trace):
-            trace["num_parameters"] = sum(map(lambda p: p.numel(), self.parameters()))
+        def append_num_parameter(job):
+            self.config.log(f"num_parameters: {sum(map(lambda p: p.numel(), self.parameters()))}")
 
-        job.post_epoch_trace_hooks.append(append_num_parameter)
+        if job.config.get("job.type") == "train":
+            job.pre_train_hooks.append(append_num_parameter)
```

Comment:
I don't like this chagne. We want models to be able to add stuff to the trace entry.

Reply:
we can always call job.config.trace()

Reply:
This method creates a new trace entry...

Reply:
The use case is: have a single valid/eval entry with all the information needed

Reply:
Ahhh, of course, we can add a hook for evaluation as well. How about pre_eval_hook and post_eval_hook?

Reply:
I think we can do this more generally using pre_run and post_run hooks, see above.